pkg/controllers/allocation/filter.go

```diff
@@ -63,6 +57,35 @@ func (f *Filter) GetProvisionablePods(ctx context.Context, provisioner *v1alpha2
 	return provisionable, nil
 }
 
+// GetProvisionerFor retrieves the provisioner responsible for the pod
+func (f *Filter) GetProvisionerFor(ctx context.Context, p *v1.Pod) (*v1alpha2.Provisioner, error) {
+	provisionerKey := client.ObjectKey{
```

Comment:
consider inlining this -- it won't be too long

Reply:
I assume you're referring to the `provisionerKey` default/default? I did it this way to make the override logic work more cleanly. I might not be seeing the alternative you're suggesting here, but if I in-line, then they'll need to be two `Get` calls and it'll get ugly IMO.

Reply:
```
provisionerKey := client.ObjectKey{ Name: "default", Namespace: "default" }
```

Reply:
over thought that one 👍   will do